cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConnectDockerfile.java

```diff
@@ -61,6 +73,57 @@ public KafkaConnectDockerfile(String fromImage, Build connectBuild) {
         writer.close();
     }
 
+    /**
+     * Generates initial stage for multi-stage build
+     * @param writer        Writer for printing the Docker commands
+     * @param plugins       List of plugins which should be added to the container image
+     */
+    private void connectorPluginsPreStage(PrintWriter writer, List<Plugin> plugins) {
+        List<Artifact> mvnArtifacts = plugins.stream().flatMap(plugin -> plugin.getArtifacts().stream()).filter(artifact -> artifact instanceof MavenArtifact).collect(Collectors.toList());
+
+        if (mvnArtifacts.size() > 0) {
+            writer.println("FROM " + mavenImage + " AS downloadArtifacts");
+            mvnArtifacts.forEach(mvn -> {
+                checkForShellMetachars(((MavenArtifact) mvn).getGroup());
+                checkForShellMetachars(((MavenArtifact) mvn).getArtifact());
+                checkForShellMetachars(((MavenArtifact) mvn).getVersion());
+                String repo = ((MavenArtifact) mvn).getRepository() == null ? MavenArtifact.DEFAULT_REPOSITORY : maybePatchRepository(((MavenArtifact) mvn).getRepository());
+                String artifactHash = Util.sha1Prefix(((MavenArtifact) mvn).getGroup() + ((MavenArtifact) mvn).getArtifact() + ((MavenArtifact) mvn).getVersion());
+                String downloadPomCmd = String.format("curl -L --create-dirs --output /tmp/%s/pom.xml %s%s/%s/%s/%s-%s.pom",
```

Comment:
You do not seem to include connector into the path. What will happen if two connectors have the same artifact? Will it not fail because of some _pre-existing_ files etc.? (same of course for the other commands)